unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Billboard/Handler/BillboardComponentHandler.cs

```diff
@@ -0,0 +1,103 @@
+﻿using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Models;
+using DCL.ECSComponents;
+using UnityEngine;
+
+namespace DCL.ECSComponents
+{
+    public class BillboardComponentHandler : IECSComponentHandler<PBBillboard>
+    {
+        private readonly IUpdateEventHandler updateEventHandler;
+        
+        private Transform entityTransform;
+        private Vector3Variable cameraPosition => CommonScriptableObjects.cameraPosition;
+        private UnityEngine.Vector3 lastPosition;
+
+        private IDCLEntity entity;
+        private IParcelScene scene;
+        private PBBillboard model;
+        
+        public BillboardComponentHandler(IUpdateEventHandler updateEventHandler)
+        {
+            this.updateEventHandler = updateEventHandler;
+            updateEventHandler.AddListener(IUpdateEventHandler.EventType.LateUpdate, LateUpdate);
+        }
+
+        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity)
+        {
+            this.entity = entity;
+            this.scene = scene;
+            
+            // Each time the camera position changes, we update the rotation of the entity
+            cameraPosition.OnChange += CameraPositionChanged;
```

Comment:
do we need this event if we are re-orienting the entity's transform in each late update?

Reply:
I think we can safely remove it, since the camera is updated in the `Update `function, removed!

Reply:
I think we can safely remove it, since the camera is updated in the `Update `function, removed!